app/src/test/java/org/simple/clinic/summary/PatientSummaryScreenControllerTest.kt

```diff
@@ -189,11 +189,23 @@ class PatientSummaryScreenControllerTest {
 
     if (wasBloodPressureSaved) {
       verify(screen).showScheduleAppointmentSheet(patientUuid)
+      verify(screen, never()).goBackToHome()
     } else {
       verify(screen).goBackToHome()
+      verify(screen, never()).showScheduleAppointmentSheet(any())
     }
   }
 
+  @Test
+  fun `when summary screen is restored, and bp was saved earlier, schedule appointment sheet should open, on clicking back or done`() {
+    uiEvents.onNext(PatientSummaryScreenCreated(patientUuid, caller = PatientSummaryCaller.SEARCH))
+    uiEvents.onNext(PatientSummaryRestoredWithBPSaved(true))
```

Comment:
It looks like we are testing one flow where `PatientSummaryRestoredWithBPSaved` is emitted with `true`, but not these remaining flows:

- emitted with false
- never emitted because `onRestoreInstanceState()` doesn't get called

What do you think?